common/lib/capa/capa/capa_problem.py

```diff
@@ -591,39 +597,37 @@ def log_error():
             # <optionresponse>
             #
             # Starting from  answer (the optioninput in this example) we go up and backwards
-            xml_elems = self.tree.xpath('//*[@id="' + answer_id + '"]')
 
-            if len(xml_elems) != 1:
+            try:
+                xml_elems = self.tree.xpath('//*[@id="' + answer_id + '"]')
+                assert len(xml_elems) == 1
```

Comment:
rather than `asserting` here. Check if the length of xml elements matches or not. in case it doesn't match return a `generate_default_question`.